src/NLPModelsIpopt.jl

```diff
@@ -2,7 +2,27 @@ module NLPModelsIpopt
 
 export ipopt
 
-using NLPModels, Ipopt
+using NLPModels, Ipopt, SolverTools
+
+const ipopt_statuses = Dict(0 => :first_order,
+                            1 => :first_order,
+                            2 => :infeasible,
+                            3 => :small_step,
+                            #4 => Diverging iterates
+                            #5 => User requestep stop
+                            #6 => Feasible point found
+                            -1 => :max_iter,
+                            #-2 => Restoration failed
+                            #-3 => Error in step computation
+                            -4 => :max_time,
+                            #-10 => Not enough degress of freedom
+                            #-11 => Invalid problem definition
+                            #-12 => Invalid option
+                            #-13 => Invalid number detected
+                            -100 => :exception,
+                            -101 => :exception,
+                            -102 => :exception,
+                            -199 => :exception)
```

Comment:
Shouldn't we grab IPOPT.jl's `ApplicationReturnStatus` and override the keys that we support more generally?

Reply:
Do you mean creating more keys in `STATUSES` in `SolverTools`?

Reply:
I meant to use their `ApplicationReturnStatus` and rename the statuses that we support (0, 1, 2, 3 and others), but keep the status that they define (which is IPOPT's) for the others (especially the last few exceptions).

Reply:
I see. However, `GenericExecutionStats` only allow status that are in `STATUSES`, so we have to choose:
- we add those to `STATUSES`;
- change `GenericExecutionStats`;
- do what I did, the translation.

Adding a few is also an option and translating others is also an option.

Notice that the message is not lost, I keep it in `solver_specific[:internal_msg]` [here](https://github.com/JuliaSmoothOptimizers/NLPModelsIpopt.jl/pull/9/files#diff-9206df8d392bada610e908fbb4c6c48bR120).

Reply:
Right. Ok, that's fine.